src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -1586,48 +1585,60 @@ bool StorageReplicatedMergeTree::checkPartChecksumsAndAddCommitOps(
     return part_found;
 }
 
+bool StorageReplicatedMergeTree::getOpsToCheckPartChecksumsAndCommit(zkutil::ZooKeeperPtr & zookeeper,
```

Comment:
We have 3 different kinds of fault injections for ZooKeeper (for different purposes), `ZooKeeperWithFaultInjection` is not my favorite one, and I'm not sure it's useful here. But I can change it here just in case

Reply:
It's not super important I'd say. I was thinking about how to help reproducing this but I think that in this case it'd be better to introduce `ThreadFuzzer::maybeInjectSleep();` after the `checkPartChecksumsAndAddCommitOps` calls to make the race condition (check in ZK vs push to ZK) more likely to happen.